### PR TITLE
Feat: Add proxy arg to client creation

### PR DESCRIPTION
**Pull request by TheCharlatan**
Allowing the user to specify a proxy, is useful to allow him to connect to a server through tor, analyze the traffic, or go through any number of anonymising services.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -193,12 +201,24 @@ pub struct RpcClient {
 
 impl RpcClient {
     /// Create a new generic RPC client that can be transformed into specialized client.
-    pub fn new(addr: String) -> Self {
-        Self {
-            inner: CallerWrapper(Arc::new(RemoteCaller {
-                http_client: reqwest::ClientBuilder::new().build().unwrap(),
-                addr,
-            })),
+    pub fn new(addr: String, proxy: Option<String>) -> Self {
```
Comment (LeoNero) [inline src/lib.rs:204]: Just to make code a bit (hopefully) cleaner, what about:
```rust
let http_client = if let Some(proxy_address) = proxy {
    reqwest::Client::builder()
        .proxy(reqwest::Proxy::all(proxy_address).unwrap())
        .build()
        .unwrap()
} else {
    reqwest::ClientBuilder::new().build().unwrap()
}

Self {
    inner: CallerWrapper(Arc::new(RemoteCaller{
        http_client,
        addr,
    }))
}
```
?

Comment (TheCharlatan) [reply src/lib.rs:204]: b0b49d5a7ef18303b6ea8e25c9964981728787ba

**Comment by LeoNero** [comment]
LGTM although clippy is being picky :P

**Comment by h4sh3d** [comment]
cACK

What about `::new(addr)` and `::with_proxy(addr, proxy)` to keep compatibility with the current API?

**Comment by TheCharlatan** [comment]
> cACK
> 
> What about `::new(addr)` and `::with_proxy(addr, proxy)` to keep compatibility with the current API?

Yeah, that makes much more sense! Done.

**Pull request closed without merging**